### PR TITLE
Various project template changes

### PR DESCRIPTION
**Pull request by matyasselmeci**
- Drop "Name"
- Make "Description" a YAML multi-line string (avoids syntax errors from `:` characters
- Add a template ResourceAllocations section

### CODE REVIEW

**Comment by matyasselmeci** [comment]
The ResourceAllocations section is nasty enough that I added the line 'please ask Software for help' since I don't expect most facilitators to dig through `/topology` to do Resource/ResourceGroup lookups. We should consider creating some sort of mapping they can use instead.

**Comment by matyasselmeci** [comment]
by mapping I mean like an endpoint for a human readable "submit node FQDN" <-> "submit node Resource Name", and also one for "HPC Site"<->"ResourceGroup"

where "HPC Site" is the shortened name of a site that we (will) tell people to put in their +Desired_Allocations list

File: template-project.yaml [hunk 1]
```diff
@@ -21,3 +22,32 @@ Sponsor:
...
   #   Name: <VO>
+
+
+### Uncomment the ResourceAllocations block if this project can make use of
+### one or more XRAC (XSEDE) or other HPC allocations.
+###
+### Contact the Software Team for assistance in filling in these values.
+# ResourceAllocations:
+
+##### Type is how the allocation was obtained: "XRAC" for allocations though
```
Comment (brianhlin) [inline template-project.yaml:33]:
```suggestion
##### Type is how the allocation was obtained: "XRAC" for allocations through
```

File: template-project.yaml [hunk 2]
```diff
@@ -21,3 +22,32 @@ Sponsor:
...
+##### XSEDE or "Other" for allocations directly through the site.
+#   - Type: Other
+
+####### SubmitResources is a list of Submit Nodes (Access Points) that users of
+####### project can submit from.  There must be at least one; each line must
```
Comment (brianhlin) [inline template-project.yaml:38]:
```suggestion
####### the project can submit from.  There must be at least one; each line must
```

**Review by brianhlin — APPROVED** [review]
LGTM

**Pull request merged**